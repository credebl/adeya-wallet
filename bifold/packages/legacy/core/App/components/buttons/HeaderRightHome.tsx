import { useNavigation } from '@react-navigation/core'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialCommunityIcons'

import { useTheme } from '../../contexts/theme'
import { Screens, TabStacks } from '../../types/navigators'
import { testIdWithKey } from '../../utils/testable'

const defaultIconSize = 26

const HeaderRightHome: React.FC = () => {
  const { t } = useTranslation()
  const navigation = useNavigation()
  const { ColorPallet } = useTheme()
  const style = StyleSheet.create({
    container: {
      marginRight: 14,
    },
  })

  return (
    <TouchableOpacity
      accessibilityLabel={t('Global.Home')}
      testID={testIdWithKey('HomeButton')}
      style={[style.container]}
      onPress={() => {
        navigation.getParent()?.navigate(TabStacks.HomeStack, { screen: Screens.Home })
      }}
    >
      <Icon name="home" size={defaultIconSize} color={ColorPallet.brand.headerIcon} />
    </TouchableOpacity>
  )
}

export default HeaderRightHome
