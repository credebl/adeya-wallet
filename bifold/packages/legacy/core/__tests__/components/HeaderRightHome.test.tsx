import { render } from '@testing-library/react-native'
import React from 'react'

import HeaderRightHome from '../../App/components/buttons/HeaderRightHome'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

describe('Header Right Home Component', () => {
  test('Renders correctly', () => {
    const tree = render(<HeaderRightHome />)

    expect(tree).toMatchSnapshot()
  })
})
