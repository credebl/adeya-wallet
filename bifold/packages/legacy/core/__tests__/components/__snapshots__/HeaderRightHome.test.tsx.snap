// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Right Home Component Renders correctly 1`] = `
<View
  accessibilityLabel="Global.Home"
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginRight": 14,
      "opacity": 1,
    }
  }
  testID="com.ariesbifold:id/HomeButton"
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 26,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ó°‹œ
  </Text>
</View>
`;
